src/main/java/net/minecraftforge/client/event/RenderTooltipEvent.java

```diff
@@ -45,20 +50,29 @@
 {
     @Nonnull
     protected final ItemStack stack;
+    @Deprecated(forRemoval = true) // TODO: remove in 1.18
     protected final List<? extends FormattedText> lines;
-    protected final PoseStack matrixStack;
+    protected final PoseStack matrixStack; // TODO: rename in 1.18
     protected int x;
     protected int y;
     protected Font fr;
+    protected final List<ClientTooltipComponent> components;
 
+    @Deprecated(forRemoval = true) // TODO: remove in 1.18
     public RenderTooltipEvent(@Nonnull ItemStack stack, @Nonnull List<? extends FormattedText> lines, PoseStack matrixStack, int x, int y, @Nonnull Font fr)
+    {
+        this(stack, matrixStack, x, y, fr, List.of());
```

Comment:
While it is a bit more of a mess, I feel like this should more or less be a copy of the new RenderTooltipEvent constructor instead of calling this, which would allow the lines parameter to be properly assigned to the lines field. This way while usage of this is deprecated it at least has proper values for the deprecated variant.